File: services/api/src/atlas_api/models/entity.py
"""Entity domain models — mirrors packages/shared/src/types/entity.ts."""

from pydantic import BaseModel, ConfigDict

from atlas_api.models.enums import EntityKind


class Entity(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    workspace_id: str
    kind: EntityKind
    name: str
    slug: str
    aliases: tuple[str, ...]
    vault_note_id: str | None
    source_ids: tuple[str, ...]
    created_at: str
    updated_at: str

File: packages/shared/src/types/entity.ts
import type { EntityId, WorkspaceId, VaultNoteId, SourceId } from "./ids.js";

export const EntityKind = {
  Person: "person",
  Company: "company",
  Project: "project",
  Paper: "paper",
  Dataset: "dataset",
  Product: "product",
} as const;
export type EntityKind = (typeof EntityKind)[keyof typeof EntityKind];

export interface Entity {
  readonly id: EntityId;
  readonly workspaceId: WorkspaceId;
  readonly kind: EntityKind;
  readonly name: string;
  readonly slug: string;
  readonly aliases: readonly string[];
  readonly vaultNoteId: VaultNoteId | null;
  readonly sourceIds: readonly SourceId[];
  readonly createdAt: string;
  readonly updatedAt: string;
}
